structuredheadings/plugin.js

```diff
@@ -51,21 +85,24 @@
       editor.addCommand("matchHeading", {
         startDisabled: true,
         exec: function () {
-          var element = editor.getSelection().getStartElement();
+          var element = getCurrentBlock(editor.getSelection().getStartElement());
           var style;
           //find previous element that matches allowedElements
-          var previousHeader = element.getPrevious(function (node) {
-            if (allowedElements.indexOf(node.getName()) >= 0) {
-              return true;
-            } else {
-              return false;
+          var previousHeader = getPreviousHeader(element);
+
+          // if already in header, set back to default
+          if (element.is(allowedElements)) {
+            switch (editor.config.enterMode) {
+            case CKEDITOR.ENTER_DIV:
+                  //eslint-disable-next-line new-cap
+              style = new CKEDITOR.style({ element: "div" });
+              break;
+            default:
+                  //eslint-disable-next-line new-cap
+              style = new CKEDITOR.style({ element: "p" });
+              element.removeClass("autonumber");
+              break;
```

Comment:
worth just writing an issue to say our default case probably doesn't work properly for ENTER_BR?